sirepo/pkcli/roles.py

```diff
@@ -5,50 +5,59 @@
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
 from __future__ import absolute_import, division, print_function
+from pykern.pkcollections import PKDict
 from pykern.pkdebug import pkdc, pkdexc, pkdlog, pkdp
 from sirepo.pkcli import admin
 import sirepo.auth
+import sirepo.auth.email
 import sirepo.auth_db
 import sirepo.server
 
 
-def add_roles(uid, *args):  # TODO(e-carlin): roles arg
+def add_roles(uid_or_email, *args):
     """Assign roles to a user
     Args:
-        uid (str): Uid of the user
+        uid_or_email (str): Uid or email of the user
         *args: The roles to assign to the user
     """
-    _check_uid_and_roles(uid, args)
-    sirepo.auth_db.UserRole.add_roles(uid, args)
-    admin.audit_proprietary_lib_files(uid)
+    u = _check_uid_or_email_and_roles(uid_or_email, args)
+    sirepo.auth_db.UserRole.add_roles(u, args)
```

Comment:
While this is probably ok, there is a problem that there's no communication with the supervisor. We add the role to sqlite, but the files aren't there. Really, sqlite should be updated after, but of course, then the auditor won't work.

We need some type of locking, but it's not important to solve right now.